frontend/public/components/factory/Table/VirtualizedTable.tsx

```diff
@@ -0,0 +1,294 @@
+  showNamespaceOverride: boolean,
+  namespace: string,
+) => {
+  const showNamespace = columnID !== 'namespace' || !namespace || showNamespaceOverride;
+  if (_.isEmpty(columns) && showNamespace) {
+    return true;
+  }
+  if (!columns.has(columnID) || !showNamespace) {
+    return false;
+  }
+  const columnIndex = [...columns].indexOf(columnID);
+  if (widthInPixels < BREAKPOINT_SM) {
+    return columnIndex < MAX_COL_XS;
+  }
+  if (widthInPixels < BREAKPOINT_MD) {
+    return columnIndex < MAX_COL_SM;
+  }
+  if (widthInPixels < BREAKPOINT_LG) {
+    return columnIndex < MAX_COL_MD;
+  }
+  if (widthInPixels < BREAKPOINT_XL) {
+    return columnIndex < MAX_COL_LG;
+  }
+  if (widthInPixels < BREAKPOINT_XXL) {
+    return columnIndex < MAX_COL_XL;
+  }
+  return true;
+};
+
+const getActiveColumns = <D extends any>(
+  windowWidth: number,
+  allColumns: TableColumn<D>[],
+  activeColumns: Set<string>,
+  columnManagementID: string,
+  showNamespaceOverride: boolean,
+  namespace: string,
+) => {
+  let columns = [...allColumns];
+  if (_.isEmpty(activeColumns)) {
+    activeColumns = new Set(
+      columns.map((col) => {
+        if (col.id && !col.additional) {
+          return col.id;
+        }
+      }),
+    );
+  }
+  if (columnManagementID) {
+    columns = columns?.filter(
+      (col) =>
+        isColumnVisible(windowWidth, col.id, activeColumns, showNamespaceOverride, namespace) ||
+        col.title === '',
+    );
+  } else {
+    columns = columns?.filter((col) => activeColumns.has(col.id) || col.title === '');
+  }
+
+  const showNamespace = !namespace || showNamespaceOverride;
+  if (!showNamespace) {
+    columns = columns.filter((column) => column.id !== 'namespace');
+  }
+  return columns;
+};
+
+export type TableColumn<D> = ICell & {
+  title: string;
+  id?: string;
+  additional?: boolean;
+  sort?: ((data: D[], sortDirection: SortByDirection) => D[]) | string;
+};
+
+export type RowProps<D, R extends any = {}> = {
+  obj: D;
+  rowData: R;
+};
+
+type VirtualizedTableProps<D, R extends any = {}> = {
+  data: D[];
+  unfilteredData: D[];
+  loaded: boolean;
+  loadError: any;
+  columns: TableColumn<D>[];
+  Row: React.ComponentType<RowProps<D, R>>;
+  NoDataEmptyMsg?: React.ComponentType<{}>;
+  EmptyMsg?: React.ComponentType<{}>;
+  scrollNode?: () => HTMLElement;
+  onSelect?: OnSelect;
+  label?: string;
+  'aria-label'?: string;
+  gridBreakPoint?: TableGridBreakpoint;
+  activeColumns?: Set<string>;
+  columnManagementID?: string;
+  showNamespaceOverride?: boolean;
+  rowData?: R;
+};
+
+const VirtualizedTable = <D, R extends any = {}>({
+  data,
+  loaded,
+  loadError,
+  columns: allColumns,
+  NoDataEmptyMsg,
+  EmptyMsg,
+  scrollNode,
+  label,
+  'aria-label': ariaLabel,
+  gridBreakPoint = TableGridBreakpoint.none,
+  onSelect,
+  Row,
+  activeColumns,
+  columnManagementID,
+  showNamespaceOverride,
+  rowData,
+  unfilteredData,
+}: VirtualizedTableProps<D, R>) => {
```

Comment:
@spadgett WDYT about typing a component like this instead of `React.FC<FooProps>` ? 
The advantage here is that when we use this component we can specify the generics like
```js
<VirtualizedTable<PodKind, PodRowData>
      {...props}
      aria-label={t('public~Machines')}
      columns={getColumns(showNodes, t)}
      Row={PodTableRow} //Row is typed as  React.ComponentType<RowProps<PodKind, PodRowData>>
      rowData={rowData} //rowData is typed as PodRowData 
    />
```

Reply:
I didn't know you could do this... I think it's more important to be able to specify generics, so this looks better to me.

As an aside, some argue [not to use `React.FC`](https://fettblog.eu/typescript-react-why-i-dont-use-react-fc/) in general.